projects/cm_mcu/I2CCommunication.c

```diff
@@ -143,24 +146,27 @@ int apollo_i2c_ctl_reg_r(uint8_t device, uint8_t address, uint8_t reg_address, u
     return *p_status;
 }
 
-int apollo_i2c_ctl_reg_w(uint8_t device, uint8_t address, uint8_t reg_address, uint8_t nbytes, int packed_data)
+int apollo_i2c_ctl_reg_w(uint8_t device, uint8_t address, uint8_t nbytes_addr, uint16_t packed_reg_address, uint8_t nbytes, uint32_t packed_data)
 {
   tSMBus* p_sMaster = pSMBus[device];
   tSMBusStatus* p_eStatus = eStatus[device];
 
   configASSERT(p_sMaster != NULL);
 
-  // first byte is the register, others are the data
-  uint8_t data[MAX_BYTES+1];
-  data[0] = reg_address;
+  // first byte (if write to one of five clcok chips) or two bytes (if write to EEPROM) is the register, others are the data
+  uint8_t data[nbytes_addr + MAX_BYTES];
+  for (int i = 0; i < nbytes_addr; ++i){
+	data[i] = (packed_reg_address >> (nbytes_addr - 1 - i) * 8) & 0xFF;
```

Comment:
fix indent here

Reply:
btw since you use vi  you might want to set "expandtab" in your vimrc, see here
https://vi.stackexchange.com/questions/4141/how-to-indent-as-spaces-instead-of-tab